### PR TITLE
Add CR consts, implement Sync() and fix queries

### PR DESCRIPTION
**Pull request by dimkouv**
- Add a package that includes exported chainReader consts. That way we can re-use them and prevent typos.
- Implement `Sync()`. It calls the offRamp to get onRamp addresses and Bind the source chain readers to them.
- Fix NextSeqNums query.

### CODE REVIEW

**Comment by dimkouv** [comment]
> ### Test Coverage
> Branch	Coverage
> dk/impl-reader-sync	67.8%
> ccip-develop	69.6%

I'll add a ticket to write integration tests using simulated backend for this implementation.

File: internal/reader/ccip.go [hunk 1]
```diff
@@ -168,26 +164,24 @@ func (r *CCIPChainReader) MsgsBetweenSeqNums(
...
-		contractName = "OffRamp"
-		funcName     = "getExpectedNextSequenceNumbers"
-	)
+	res := make([]cciptypes.SeqNum, 0, len(chains))
+	for _, chain := range chains {
+		cfg, exists := cfgs[chain]
+		if !exists {
+			return nil, fmt.Errorf("source chain config not found for chain %d", chain)
+		}
+		if cfg.MinSeqNr == 0 {
```
Comment (asoliman92) [inline internal/reader/ccip.go:178]: What if it's first message ever on a new chain?

File: internal/reader/ccip.go [hunk 2]
```diff
@@ -216,13 +210,79 @@ func (r *CCIPChainReader) GasPrices(ctx context.Context, chains []cciptypes.Chai
...
+	for chain, cfg := range sourceConfigs {
+		if cfg.OnRamp == "" {
+			return false, fmt.Errorf("onRamp address not found for chain %d", chain)
+		}
+
+		// Bind the onRamp contract address to the reader.
+		// If the same address exists -> no-op
+		// If the address is changed -> updates the address, overwrites the existing one
+		// If the contract not binded -> binds to the new address
+		if err := r.contractReaders[chain].Bind(ctx, []types.BoundContract{
```
Comment (asoliman92) [inline internal/reader/ccip.go:227]: Can we check if it's an update and log? Can be helpful for debugging purposes.

File: internal/reader/ccip.go (hunk 1 above)
Comment (dimkouv) [reply internal/reader/ccip.go:179]: next expected is 1

File: internal/reader/ccip.go (hunk 2 above)
Comment (dimkouv) [reply internal/reader/ccip.go:228]: no we can't unfortunately

File: internal/reader/ccip.go (hunk 1 above)
Comment (asoliman92) [reply internal/reader/ccip.go:179]: Can we already change names?! 😂 😮‍💨

**Review by asoliman92 — APPROVED** [review]

File: internal/reader/ccip.go (hunk 1 above)
Comment (dimkouv) [reply internal/reader/ccip.go:179]: i'll fix in another pr

**Pull request merged**